common/flatpak-utils-http.c

```diff
@@ -231,6 +243,159 @@ check_http_status (guint status_code,
   return FALSE;
 }
 
+FlatpakCertificates*
+flatpak_get_certificates_for_uri(const char *uri,
+                                 GError    **error)
+{
+  g_autoptr(FlatpakCertificates) certificates = NULL;
+  g_autoptr(GUri) parsed_uri = NULL;
+  g_autofree char *hostport = NULL;
+  const char *system_certs_d = NULL;
+  g_autofree char *certs_path_str = NULL;
+  g_auto(GStrv) certs_path = NULL;
+
+  parsed_uri = g_uri_parse (uri, G_URI_FLAGS_PARSE_RELAXED, error);
+  if (!parsed_uri)
+    return NULL;
+
+  if (!g_uri_get_host (parsed_uri))
+    return NULL;
+
+  if (g_uri_get_port (parsed_uri) != -1)
+    hostport = g_strdup_printf ("%s:%d", g_uri_get_host(parsed_uri), g_uri_get_port(parsed_uri));
+  else
+    hostport = g_strdup (g_uri_get_host(parsed_uri));
+
+  system_certs_d = g_getenv ("FLATPAK_SYSTEM_CERTS_D");
+  if (system_certs_d == NULL || system_certs_d[0] == '\0')
+    system_certs_d = "/etc/containers/certs.d:/etc/docker/certs.d";
+
+  /* containers/image hardcodes ~/.config and doesn't honor XDG_CONFIG_HOME */
```

Comment:
Shouldn't we do that as well then? Or at least look in both?

Reply:
I think it's a bug in containers/image that will get fixed at some point since the libpod stack (including containers/image) uses $XDG_CONFIG_HOME in many places, but just not this - e.g. it looks for $XDG_CONFIG_HOME/containers/auth.json. I guess filing PR for containers/image would be the way to make that certain.